src/pages/Courses/CourseEditor.tsx

```diff
@@ -44,37 +45,80 @@ const validationSchema = Yup.object({
 });
 
 const CourseEditor: React.FC<IEditor> = ({ mode }) => {
-
   // API hook for making requests
   const { data: courseResponse, error: courseError, sendRequest } = useAPI();
+  const { data: users, sendRequest: fetchusers } = useAPI();
   const auth = useSelector(
     (state: RootState) => state.authentication,
     (prev, next) => prev.isAuthenticated === next.isAuthenticated
   );
-  const { courseData, institutions, instructors }: any = useLoaderData();
+  const { courseData, institutions }: any = useLoaderData();
   const dispatch = useDispatch();
   const navigate = useNavigate();
   const location = useLocation();
 
-  initialValues.institution_id = auth.user.institution_id;
+  interface IFormOption {
+    label: string;
+    value: string;
+  }
+
+  const [filteredInstructors, setFilteredInstructors] = useState<IFormOption[]>([]);
+  const [selectedInstitutionId, setSelectedInstitutionId] = useState<number | null>(null); // New state for selected institution
 
-  // Close the modal if the course is updated successfully and navigate to the courses page
   useEffect(() => {
-    if (courseResponse && courseResponse.status >= 200 && courseResponse.status < 300) {
-      dispatch(
-        alertActions.showAlert({
-          variant: "success",
-          message: `Course ${courseData.name} ${mode}d successfully!`,
-        })
-      );
-      navigate(location.state?.from ? location.state.from : "/courses");
-    }
-  }, [dispatch, mode, navigate, courseData.name, courseResponse, location.state?.from]);
+    fetchusers({url:'/users'});
+  }, [fetchusers]);
 
-  // Show the error message if the course is not updated successfully
+  // Filter instructors based on selected institution
   useEffect(() => {
-    courseError && dispatch(alertActions.showAlert({ variant: "danger", message: courseError }));
-  }, [courseError, dispatch]);
+    
+    if (users) {
+      const instructorsList: IFormOption[] = [{ label: 'Select an Instructor', value: '' }];
+      console.log('Selected Institution ID:', selectedInstitutionId)
+     
+      // Filter by instructors by institution
+      const onlyInstructors = users.data.filter((user: any) => 
+        (user.role.name === 'Instructor')&& (user.institution.id === selectedInstitutionId)); 
+      console.log('Users:', users.data)
+      onlyInstructors.forEach((instructor: any) => {
+        instructorsList.push({ label: instructor.name, value: String(instructor.id) });
+      });
+      setFilteredInstructors(instructorsList);
+      
+    }
+  }, [users, selectedInstitutionId]); // Re-run this effect when users or selectedInstitutionId changes
+  
+  // Handle institution selection change and implement Single Responsibility Principle
+const handleInstitutionChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+  const institutionId = Number(event.target.value);
+  setSelectedInstitutionId(institutionId);
+};
+// Success handler for course submission
```

Comment:
In CourseEditor.tsx, the handleCourseSuccess and handleCourseError functions have been separated to enhance readability and follow the Single Responsibility Principle (SRP).